src/compiler/transformers/decorators-to-static/import-alias-map.ts

```diff
@@ -0,0 +1,42 @@
+import ts from 'typescript';
+
+import { STENCIL_DECORATORS, StencilDecorator } from './decorators-constants';
+
+export class ImportAliasMap extends Map<StencilDecorator, string> {
+  constructor(sourceFile: ts.SourceFile) {
+    super();
+    this.generateImportAliasMap(sourceFile);
+  }
+
+  /**
+   * Parses a {@link ts.SourceFile} and generates a map of all imported Stencil decorators
+   * to their aliases import name (if one exists).
+   *
+   * @param sourceFile The source file to parse
+   */
+  private generateImportAliasMap(sourceFile: ts.SourceFile) {
+    const importDeclarations = sourceFile.statements.filter(ts.isImportDeclaration);
+
+    for (const importDeclaration of importDeclarations) {
+      if (importDeclaration.moduleSpecifier.getText().includes('@stencil/core')) {
+        const namedBindings = importDeclaration.importClause?.namedBindings;
+
+        if (namedBindings && ts.isNamedImports(namedBindings)) {
+          for (const element of namedBindings.elements) {
+            const importName = element.name.getText();
+            const originalImportName = element.propertyName?.getText() ?? importName;
+
+            // We only care to generate a map for the Stencil decorators
+            if (STENCIL_DECORATORS.includes(originalImportName as StencilDecorator)) {
+              this.set(originalImportName as StencilDecorator, importName);
+            }
+          }
+        }
+      }
+    }
+  }
+
+  override get(key: StencilDecorator): string {
+    return super.get(key) ?? key;
```

Comment:
Is there a valid use case to ask for a decorator key we don't know?

Reply:
This isn't for the case of a decorator we don't know, but for a decorator that wasn't used in the component. If we don't override this, when we pass decorator names into the functions responsible for transforming them, we'd need to do `importAliasMap.get('Watch') ?? 'Watch'`. Since if the component didn't have a watch function, the map would not have a key for `Watch` and would return `undefined`

Reply:
We could, alternatively, change the class so that it always has all the Stencil decorators and we just change the value for any aliased keys, works the same.

Reply:
Thanks for clarifying. This makes sense to me and can stay as is.